### PR TITLE
 Closes #99 - Create unit tests for Preset controller

### LINKED ISSUE #99: Create unit tests for Preset controller

### PR DESCRIPTION
**Pull request by oleksandrlytvyn1223**

### CODE REVIEW

File: Common/Exceptions/IncorrectPasswordException.cs [hunk 1]
```diff
@@ -5,9 +5,9 @@ namespace Common.Exceptions
 {
     public class IncorrectPasswordException : Exception
```
Comment (roman-fay) [inline Common/Exceptions/IncorrectPasswordException.cs]: didn't we agree to rename this to PasswordMismatchException?

File: Common/Exceptions/StandartPresetException.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace Common.Exceptions
+{
+    public class StandartPresetException : Exception
```
Comment (roman-fay) [inline Common/Exceptions/StandartPresetException.cs]: bad name. when this exception might be thrown?

File: ServiceTier/Services/IPresetService.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@ public interface IPresetService : IBaseService
         void UpdatePreset(PresetDTO presetDTO);
         void DeletePresetById(int presetid);
         PresetDTO GetPresetById(int presetid);
-        IQueryable<PresetDTO> GetAllStandardPresets();
-        IQueryable<PresetDTO> GetAllCustomPresetsByUserId(int userid);
+        List<PresetDTO> GetAllStandardPresets();
```
Comment (roman-fay) [inline ServiceTier/Services/IPresetService.cs]: using interface here is better (IList, maybe even ICollection)

File: ServiceTier/Services/PresetService.cs [hunk 4]
```diff
@@ -80,18 +76,14 @@ public IQueryable<PresetDTO> GetAllCustomPresetsByUserId(int userid)
...
-            return listOfPresetsDTO.AsQueryable();
+            return listOfPresetsDTO;
         }
 
-        public IQueryable<PresetDTO> GetAllStandardPresets()
+        public List<PresetDTO> GetAllStandardPresets()
         {
             var listOfPresetsDTO = new List<PresetDTO>();
             var presets = unitOfWork.Presets.GetAll();
             var timers = unitOfWork.Timers.GetAll();
```
Comment (roman-fay) [inline ServiceTier/Services/PresetService.cs]: I guess new repo methods are not implemented yet, right?

File: Tests/ServiceTierTests/PresetsServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            unitOfWork = new Mock<IUnitOfWork>();
+            timerService = new Mock<ITimerService>();
+            subject = new PresetService(unitOfWork.Object,timerService.Object);
+        }
+
+        [Test]
+        public void CreatePreset()
+        {
+            List<TimerDTO> timers = new List<TimerDTO>();
+            Preset preset = new Preset { Name="preset", UserId=315 };
```
Comment (roman-fay) [inline Tests/ServiceTierTests/PresetsServiceTests.cs]: make 315 as a const

Comment (roman-fay) [reply Tests/ServiceTierTests/PresetsServiceTests.cs]: also"preset" (and everything that is used more than once in the tests)

File: ServiceTier/Services/PresetService.cs [hunk 6]
```diff
@@ -39,11 +40,11 @@ public void DeletePresetById(int presetid)
         {
```
Comment (roman-fay) [inline ServiceTier/Services/PresetService.cs]: again - what if user doesn't own this preset. do we handle (or plan to) this scenario somehow?

File: Common/Exceptions/StandartPresetException.cs (hunk 2 above)
Comment (oleksandrlytvyn1223) [reply Common/Exceptions/StandartPresetException.cs]: when user tries to delete a standart preset

**Pull request merged**